src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeReader.cs

```diff
@@ -259,6 +259,84 @@ public override async ValueTask<ReadResult> ReadAsync(CancellationToken cancella
             }
         }
 
+        /// <inheritdoc />
+        public override async Task CopyToAsync(PipeWriter destination, CancellationToken cancellationToken = default)
+        {
+            ThrowIfCompleted();
+
+            // PERF: store InternalTokenSource locally to avoid querying it twice (which acquires a lock)
+            CancellationTokenSource tokenSource = InternalTokenSource;
+            if (TryReadInternal(tokenSource, out ReadResult readResult))
+            {
+                await WriteAsyncInternal(
+                    destination,
+                    (destination, memory, cancellationToken) => destination.WriteAsync(memory, cancellationToken),
+                    readResult,
+                    cancellationToken).ConfigureAwait(false);
+            }
+
+            if (_isStreamCompleted) return;
+
+            CancellationTokenRegistration reg = default;
+            if (cancellationToken.CanBeCanceled)
+            {
+                reg = cancellationToken.UnsafeRegister(state => ((StreamPipeReader)state!).Cancel(), this);
+            }
+
+            using (reg)
+            {
+                try
+                {
+                    await InnerStream.CopyToAsync(destination, tokenSource.Token).ConfigureAwait(false);
+                }
+                catch (OperationCanceledException)
+                {
+                    ClearCancellationToken();
+
+                    throw;
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public override async Task CopyToAsync(Stream destination, CancellationToken cancellationToken = default)
+        {
+            ThrowIfCompleted();
+
+            // PERF: store InternalTokenSource locally to avoid querying it twice (which acquires a lock)
+            CancellationTokenSource tokenSource = InternalTokenSource;
+            if (TryReadInternal(tokenSource, out ReadResult readResult))
+            {
+                await WriteAsyncInternal(destination, WriteAsyncFunc(), readResult, cancellationToken).ConfigureAwait(false);
+            }
+
+            if (_isStreamCompleted) return;
+
+            CancellationTokenRegistration reg = default;
+            if (cancellationToken.CanBeCanceled)
+            {
+                reg = cancellationToken.UnsafeRegister(state => ((StreamPipeReader)state!).Cancel(), this);
+            }
+
+            using (reg)
+            {
+                try
+                {
+#if (!NETSTANDARD2_0 && !NETFRAMEWORK)
+                    await InnerStream.CopyToAsync(destination, tokenSource.Token).ConfigureAwait(false);
+#else
+                    await InnerStream.CopyToAsync(destination, 81920, tokenSource.Token).ConfigureAwait(false);
```

Comment:
I know the base Stream implementation does this by why aren't we just delegating to the other overload here?

Reply:
AFAIK the `Stream.CopyToAsync(Stream, CancellationToken)`  overload has been introduced in netstandard2.1, here I must provide a `buffersize` 😞
Do I miss something here?

Reply:
Why do you need to call it? Just call the one without a buffer size.

Reply:
The overload without a buffer size does not exist when targeting the `net461` and `netstandard2.0` TFMs.

Reply:
https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.copytoasync?view=net-5.0#System_IO_Stream_CopyToAsync_System_IO_Stream_System_Threading_CancellationToken_

It exists everywhere...

Reply:
I get a build error for `net461` and `netstandard2.0`:
![image](https://user-images.githubusercontent.com/2341261/116794270-0a6e8e80-aacc-11eb-8b52-8b111b83f361.png)

Reply:
OK leave it for now. I'll look into it.

Reply:
Ah I figured it out. The overload with both the token *and* the Stream is only on ns2.1. Can you do me a favor and move this code to 

https://github.com/dotnet/runtime/blob/main/src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/StreamExtensions.netstandard.cs

I'd like to remove all ifdefs from product code into these shims.

Reply:
Favor granted! ^^